syntax = "proto3";

package messenger;
option go_package = "example.com/messenger";
import "google/protobuf/empty.proto";

service MessengerService {
  rpc SendMessage (MessageRequest) returns (MessageResponse);
  rpc GetMessages (GetMessagesRequest) returns (GetMessagesResponse);
}

service EncryptionService {
  rpc GetClientId (google.protobuf.Empty) returns (Id);
  rpc GetRsaKey (Id) returns (RsaKey);                   
  rpc GiveRsaKey (RsaKey) returns (google.protobuf.Empty); 
}

message MessageRequest {
  int32 id = 1;
  string user = 2;
  bytes encrypted_content = 3;
}

message Id {
  int32 id = 1;
}

message RsaKey {
  int32 id = 1;
  bytes RsaPublicKey = 2;
}

message MessageResponse {
  bool success = 1;
}

message GetMessagesRequest {
  int32 id = 1;
}

message GetMessagesResponse {
  repeated Message messages = 1;
}

message Message {
  int32 id = 1;
  string user = 2;
  string content = 3;
  string timestamp = 4;
}
